src/hospitalizations.rs

```diff
@@ -0,0 +1,399 @@
+use ixa::{
+    define_data_plugin, define_person_property_with_default, define_rng, trace, Context,
+    ContextPeopleExt, ContextRandomExt, HashMap, PersonId, PersonPropertyChangeEvent,
+};
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    infectiousness_manager::InfectionStatusValue,
+    interventions::ContextTransmissionModifierExt,
+    parameters::ContextParametersExt,
+    population_loader::Age,
+    symptom_progression::{SymptomValue, Symptoms},
+};
+
+define_person_property_with_default!(Hospitalized, bool, false);
+
+define_rng!(HospitalizationRng);
+
+#[derive(Debug, Clone, Copy, Deserialize, Serialize, PartialEq)]
+
+pub struct HospitalAgeGroups {
+    pub min: u8,
+    pub probability: f64,
+}
+
+#[derive(Default)]
+struct HospitalDataContainer {
+    age_group_mapping: HashMap<u8, HospitalAgeGroups>,
+}
+
+impl HospitalDataContainer {
+    fn set_age_group_mapping(&mut self, age_groups: &[HospitalAgeGroups]) {
+        for age in 0..=120 {
+            for i in 1..age_groups.len() {
+                let group = &age_groups[i - 1];
+                let next_group = age_groups[i];
+                if (group.min..next_group.min).contains(&age) {
+                    self.age_group_mapping.insert(age, *group);
+                    break;
+                } else if age >= age_groups.last().unwrap().min {
+                    self.age_group_mapping
+                        .insert(age, *age_groups.last().unwrap());
+                    break;
+                }
+            }
+        }
+    }
+
+    fn get_age_group(&self, age: u8) -> &HospitalAgeGroups {
+        self.age_group_mapping
+            .get(&age)
+            .expect("Age group not found")
+    }
+}
+
+define_data_plugin!(
+    HospitalDataPlugin,
+    HospitalDataContainer,
+    HospitalDataContainer::default()
+);
+
+trait ContextHospitalizationInternalExt {
+    fn plan_hospital_arrival(&mut self, person_id: PersonId) -> Result<(), ixa::IxaError>;
+    fn plan_hospital_departure(&mut self, person_id: PersonId) -> Result<(), ixa::IxaError>;
+    fn setup_hospitalization_event_sequence(&mut self);
+    fn evaluate_hospitalization_risk(&mut self, person_id: PersonId) -> bool;
+    fn set_age_group_mapping(&mut self);
+}
+
+impl ContextHospitalizationInternalExt for Context {
```

Comment:
Use the `PluginContext` syntax as in #222:
```suggestion
trait ContextHospitalizationInternalExt: PluginContext + ContextRadnomExt + ContextPeople Ext{
```
followed by the `impl` block:
```
impl ContextHospitalizationInternalExt for Context {}
```